src/Features/Core/Portable/MakeClassAbstract/AbstractMakeClassAbstractCodeFixProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.MakeClassAbstract
+{
+    internal abstract class AbstractMakeClassAbstractCodeFixProvider<TClassDeclarationSyntax> : SyntaxEditorBasedCodeFixProvider
+        where TClassDeclarationSyntax : SyntaxNode
+    {
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.Compile;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            if (context.Diagnostics.Length == 1 &&
+                IsValidMemberNode(context.Diagnostics[0].Location?.FindNode(context.CancellationToken)))
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics[0], c)),
+                    context.Diagnostics);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected sealed override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor,
+            CancellationToken cancellationToken)
+        {
+            for (var i = 0; i < diagnostics.Length; i++)
+            {
+                var declaration = diagnostics[i].Location?.FindNode(cancellationToken);
+
+                if (IsValidMemberNode(declaration))
+                {
+                    var enclosingClass = declaration.FirstAncestorOrSelf<TClassDeclarationSyntax>();
+                    editor.ReplaceNode(enclosingClass,
+                        (currentClassDeclaration, generator) => generator.WithModifiers(currentClassDeclaration, DeclarationModifiers.Abstract));
```

Comment:
won't this blow away other modifiers?  don't you need to 'union' this with the existing mods?

Reply:
That was my thought at first but this seem to keep the existing modifiers and just apply the new one (with some validity check).

Reply:
Yes, the current behavior seems strange to me for a With method. Ideally it should overwrite, not merge. It is unlikely we would change the behavior though as it would be a breaking change.

Reply:
@mavasani Well it can simply be overwrite as we need to keep the other modifiers or am I missing something?

Reply:
@mavasani so what do you want me to do here?